calculator.go

```diff
@@ -1,13 +1,70 @@
 // Package calculator provides a library for simple calculations in Go.
 package calculator
 
+import (
+	"errors"
+	"fmt"
+	"math"
+)
+
 // Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-return a + b
+func Add(a, b float64, nums ...float64) float64 {
+	total := a + b
+	if len(nums) > 0 {
+		for _, d := range nums {
+			total += d
+		}
+	}
+	return total
 }
 
 // Subtract takes two numbers and returns the result of subtracting the second
 // from the first.
-func Subtract(a, b float64) float64 {
-	return b - a
+func Subtract(a, b float64, nums ...float64) float64 {
+	total := a - b
+	if len(nums) > 0 {
+		for _, d := range nums {
+			total -= d
+		}
+	}
+	return total
+}
+
+// Multiply takes two numbers and returns the result of multiplying the first by
+// the second.
+func Multiply(a, b float64, nums ...float64) float64 {
+	total := a * b
+	if len(nums) > 0 {
+		for _, d := range nums {
+			total *= d
+		}
+	}
+	return total
+
+}
+
+// Divide takes two numbers and returns the result of dividing the first by the
+// second, or an error if the second value is zero.
+func Divide(a, b float64, nums ...float64) (float64, error) {
+	if b == 0 {
+		return 0, errors.New("division by zero is undefined")
+	}
+	total := a / b
+	if len(nums) > 0 {
```

Comment:
We could actually skip this check, couldn't we? If there are zero elements in `nums` then the `for... range` loop will execute zero times, which is already correct.